### ISSUE TITLE
node.loc.source is undefined

### ISSUE THREAD

**Issue by olsonpm** [eslint/eslint#5033]
According to the [estree documentation of the node object](https://github.com/estree/estree/blob/master/spec.md#node-objects)

```
interface Node {
    type: string;
    loc: SourceLocation | null;
}  

interface SourceLocation {
    source: string | null;
    start: Position;
    end: Position;
}
```
This plugin code should print "false"

```javascript
function arrayIsArray(ctx) {
  return {
    MemberExpression: function(node) {
      console.log("source is undefined? " + (typeof node.loc.source === 'undefined'));
...

// prints: "source is undefined? true"
```

I didn't look into the code enough to know whether this repo was the correct place to file the issue.  estree is a spec and not an implementation, and I didn't look into what implementation eslint uses.

side note: the [eslint documentation here](http://eslint.org/docs/developer-guide/working-with-rules#rule-basics) should link to the implementation instead of the spec

Edit: eslint version "1.10.3"

**Comment by eslintbot** [comment]
@olsonpm Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
That seems like a spec bug to me. As far as I'm aware, there's no implementation that includes the `source` property.

**Comment by olsonpm** [comment]
Interesting.  If you're confident in that, I will submit a bug there.  I'm sorry I just don't have time to look into this very deeply.  I'm very new to all of this and am creating an IE8 linter for work.

**Comment by nzakas** [comment]
Yeah, I've been staring at these ASTs for years, and I've just never seen it. You can try yourself at astexplorer.net, as well. My guess is that there might have been one parser that did it at one time and so it was included, I'm just not aware of any now (at least, JavaScript-based ones).

**Comment by olsonpm** [comment]
I see.  Thanks much for explaining - it really helps.
